File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.py
class Solution:
    def numberOfWeakCharacters(self, properties: List[List[int]]) -> int:
        properties.sort(key= lambda x: (x[0], -x[1]))
        
        count = 0
        minm = float("-inf")
        
        for v in range(len(properties)-1, -1, -1):
            if(properties[v][1]<minm):
                count+=1
            
            minm = max(properties[v][1], minm)
        
        return count

File: 1996-the-number-of-weak-characters-in-the-game/1996-the-number-of-weak-characters-in-the-game.js
/**
 * @param {number[][]} properties
 * @return {number}
 */
var numberOfWeakCharacters = function(properties) {
    properties.sort((a, b) => {
        if(a[0] === b[0]){
            return b[1]-a[1]
        }
        return a[0] - b[0]
    })
    console.log(properties)
    let count = 0
    let min = -Infinity
    for(let i=properties.length-1; i>=0; i--){
        if(properties[i][1]<min){
            count++          
        }
        min = Math.max(min, properties[i][1])
    }
    
    return count
};